Jenkinsfile

```diff
@@ -158,27 +158,57 @@ pipeline {
                 )
             }
         }
-        stage("Create a 'dev' and 'latest' tag from 'master'") {
+        stage("Create a 'latest' tag from 'master'") {
             when {
                 branch 'master'
             }
-            environment {
-                // qe-pipeline-library step
-                DOCKER_DEV_TAG = getDevTag("${env.CLONED_REPOSITORY_PATH}")
-            }
             steps{
                 // qe-pipeline-library step
                 tagRemoteContainerImage(
                     credentialsId: "${env.CREDENTIALS_ID}",
                     sourceImage: "${env.OPERATOR_CONTAINER_IMAGE_NAME}",
-                    targetImage: "quay.io/aerogear/${env.OPERATOR_NAME}:${env.DOCKER_DEV_TAG}",
+                    targetImage: "${env.OPERATOR_CONTAINER_IMAGE_NAME_latest}",
                     deleteOriginalImage: false
                 )
+            }
+        }
+        // rebuild and push the image so that it points to master tag of operand (for nightly testing purposes)
+        stage("Rebuild and push 'dev' image for nightly testing") {
+            when {
+                branch 'master'
+            }
+            environment {
+                // qe-pipeline-library step
+                DOCKER_DEV_TAG = getDevTag("${env.CLONED_REPOSITORY_PATH}")
+            }
+            steps{
+                dir("${env.CLONED_REPOSITORY_PATH}") {
+                    // remove original build
+                    sh "rm -rf build/_output"
+
+                    // replace specific operand version with 'master'
+                    sh 'sed -i -E \'s/"MDC_IMAGE_STREAM_TAG", "[a-zA-Z0-9.+-]*"/"MDC_IMAGE_STREAM_TAG", "master"/\' pkg/config/config.go'
+                    sh 'sed -i -E \'s/quay.io\\/aerogear\\/mobile-developer-console:[a-zA-Z0-9.+-]*/quay.io\\/aerogear\\/mobile-developer-console:master/\' pkg/config/config.go'
+
+                    // compile
+                    sh "make code/compile"
+
+                    // build image with full dev tag
+                    // qe-pipeline-library step
+                    dockerBuildAndPush(
```

Comment:
It might be nice if this step accepted a map of additional labels to add to the image. This way you could have a label for the commit that the image is built from (and you wouldn't have to have two tags), and you could also have something like `quay.expires-after=2w` so that these dev images (and even the PR images) get expired.

Reply:
That makes sense, will add the optional map. For the `expires-after`, I understand it like that it needs to be specified in Dockerfile via label. Or can it be set for existing images before/when they are pushed?

Reply:
It's the same as any other label (like the other label I'm suggesting for git commit hash). They can be specified by passing `--label` params to docker/podman.

See in the makefile how we pass it through to docker/podman from the `operator-sdk build` command: https://github.com/aerogear/mobile-developer-console-operator/blob/master/Makefile#L84

Reply:
Ok, do I understand it right, that there would be just one `dev` tag in quay.io? Labeled with commit info and with every push would be republished? So there would not be previous dev/testing images?

If so, and we would label it with the `expires-after`, we could lose testing image if there were no updates to master of operator for some time.

Reply:
Yeah, you're right, the `expires-after` is probably not appropriate for _this_ PR, but separately would be useful for PR builds, if additional labels could be passed through the library function.